horace_core/algorithms/private/gen_sqw_check_files.m

```diff
@@ -99,43 +99,122 @@
+            % only one copy of the struct will actually be in memory
+            par_file_out = repmat({strtrim(par_file{1})},n_par_files,1);
+        else
+            error('HORACE:gen_sqw_check_files:invalid_argument', ...
+                  'wrong type of contents of cell of one par file');
         end
-        if ~exist(par_file_out,'file')
-            error('HORACE:algorithms:invalid_argument',...
-                'Detector parameter file "%s" not found',par_file_out);
+    elseif numel(par_file) == n_par_files
+        if cellfun(@(x)istext(x),par_file)
+            % cell array with text element == nr spe files has been provided
+            % use it as is trimmed
+            par_file_out = cellfun(@strtrim,par_file,'UniformOutput',false);            
+            det_par_file = true(n_par_files,1);
+        elseif cellfun(@(x)isstruct(x),par_file)
+            % cell array with N struct elements has been provided
+            % par file out is filled with these elements
+            det_par_file = true(n_par_files,1);
+            % all elements copied
+            % this may not be memory efficient 
+            par_file_out = par_file; 
+        else
+            error('HORACE:gen_sqw_check_files:invalid_argument', ...
+                  'wrong type of contents of cell of N par files');
         end
     else
-        pf = {'filename','filepath','group','x2','phi','azim','width','height'};
-        if ~all(isfield(par_file,pf))
-            error('HORACE:algorithms:invalid_argument',...
-                'Detector parameter information provided as input structure must be in Horace par_file format');
-        end
-        par_file_out = par_file.filename;
+        error('HORACE:gen_sqw_check_files:invalid_argument', ...
+                  'cell array of par files must be all filenames or all structs');
     end
 else
-    det_par_file = false;
-    par_file_out='';
+    % other possible inputs are errors
+    error('HORACE:algorithms:invalid_argument', ...
+          'not acceptable input format for par files');
 end
 
+% define the fields for a .par struct
+pf = {'filename','filepath','group','x2','phi','azim','width','height'};
+
+% Check the par file names and take copies (copy or struct filename) for
+% the name clash check following
+par_file_names = cell(n_par_files,1);
+for ii=1:n_par_files
+    pfile = par_file_out{ii};
+    if ~isempty(pfile)
+    % parameter file item has been given        
+        % if the .par file name has been input, check for existence
+        if det_par_file(ii)
+            % Check par file has legal extension and exists
+            [~,~,ext]=fileparts(par_file_out{ii});
+            if any(strcmpi(ext,[ext_horace,'.spe']))
+                error('HORACE:algorithms:invalid_argument',...
+                    ['Detector parameter files must not have the reserved extension: "%s". ',...
+                    '''. Check the file is .par type and rename.'],ext);
```

Comment:
You say `par` here, but `spe` above

Reply:
ext_horace is the cellstr of horace extensions on L.34. This adds to that cellstr rather than concatenating as ext_horace is a cell. .par types should be reexpressed without the . so we don't think it's an extension rather than a file type